File: 169-majority-element/169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& a) {
       int n=a.size();
        int c=0;
        int can;
        for(int i=0;i<n;i++)
        {
            if(c==0)
            {
                can=a[i];
            }
            if(can==a[i])
            {
                c++;
            }
            else
            {
                c--;
            }
        }
        return can;
    }
    
};

File: 169-majority-element/169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int cur=0;
        int cnt=0;
        int n = nums.length;
        for(int i=0;i<n;i++){
            if(cnt == 0){
                cur = nums[i];
            }
            if(cur == nums[i]){
                cnt +=1;
            }else{
                cnt -= 1;
            }
        }
        return cur;
    }
}